core/field_colour_slider.js

```diff
@@ -0,0 +1,346 @@
+    this.brightnessReadout_.textContent = Math.floor(100 * hsv[2] / 255).toFixed(0);
+  }
+};
+
+/**
+ * Update the slider handle positions
+ */
+Blockly.FieldColourSlider.prototype.updateSliderHandles_ = function() {
+  if (this.hueSlider_) {
+    var hsv = goog.color.hexToHsv(this.getValue());
+    this.hueSlider_.animatedSetValue(hsv[0]);
+    this.saturationSlider_.animatedSetValue(hsv[1]);
+    this.brightnessSlider_.animatedSetValue(hsv[2]);
+  }
+};
+
+/**
+ * Get the text from this field.  Used when the block is collapsed.
+ * @return {string} Current text.
+ */
+Blockly.FieldColourSlider.prototype.getText = function() {
+  var colour = this.colour_;
+  // Try to use #rgb format if possible, rather than #rrggbb.
+  var m = colour.match(/^#(.)\1(.)\2(.)\3$/);
+  if (m) {
+    colour = '#' + m[1] + m[2] + m[3];
+  }
+  return colour;
+};
+
+/**
+ * Create label and readout DOM elements, returning the readout
+ * @param {string} labelText - Text for the label
+ * @return {Array} The container node and the readout node.
+ * @private
+ */
+Blockly.FieldColourSlider.prototype.createLabelDom_ = function(labelText) {
+  var labelContainer = document.createElement('div');
+  labelContainer.setAttribute('class', 'scratchColourPickerLabel');
+  var readout = document.createElement('span');
+  readout.setAttribute('class', 'scratchColourPickerReadout');
+  var label = document.createElement('span');
+  label.setAttribute('class', 'scratchColourPickerLabelText');
+  label.innerHTML = labelText;
+  labelContainer.appendChild(label);
+  labelContainer.appendChild(readout);
+  return [labelContainer, readout];
+};
+
+/**
+ * Factory for creating the different slider callbacks
+ * @param {string} channel - One of "hue", "saturation" or "brightness"
+ * @return {function} the callback for slider update
+ */
+Blockly.FieldColourSlider.prototype.sliderCallbackFactory = function(channel) {
+  var thisField = this;
+  return function(event) {
+    var channelValue = event.target.getValue();
+    var hsv = goog.color.hexToHsv(thisField.getValue());
+    switch (channel) {
+      case 'hue':
+        hsv[0] = channelValue;
+        break;
+      case 'saturation':
+        hsv[1] = channelValue;
+        break;
+      case 'brightness':
+        hsv[2] = channelValue;
+        break;
+    }
+    var colour = goog.color.hsvToHex(hsv[0], hsv[1], hsv[2]);
+    if (thisField.sourceBlock_) {
+         // Call any validation function, and allow it to override.
+      colour = thisField.callValidator(colour);
+    }
+    if (colour !== null) {
+      thisField.setValue(colour, true);
+    }
+  };
+};
+
+/**
+ * Activate the eyedropper, passing in a callback for setting the field value.
+ * @private
+ */
+Blockly.FieldColourSlider.prototype.activateEyedropperInternal_ = function() {
+  var thisField = this;
+  Blockly.FieldColourSlider.activateEyedropper_(function(value) {
+    thisField.setValue(value);
+  });
+};
+
+/**
+ * Create a palette under the colour field.
```

Comment:
Update this comment to reflect new behavior

Reply:
Fixed 5f8184d